src/vector/geometry.rs

```diff
@@ -424,10 +426,38 @@ impl Geometry {
         Ok(unsafe { Geometry::with_c_geometry(new_c_geom, true) })
     }
 
+    /// Compute geometry area in square units of the spatial reference system in use.
+    ///
+    /// Supported for `LinearRing`, `Polygon` or `MultiPolygon`.
+    /// Returns zero for all other geometry types.
+    ///
+    /// See: [`OGR_G_Area`](https://gdal.org/api/vector_c_api.html#_CPPv410OGR_G_Area12OGRGeometryH)
     pub fn area(&self) -> f64 {
         unsafe { gdal_sys::OGR_G_Area(self.c_geometry()) }
     }
 
+    /// Computes and returns the axis-aligned 2D bounding envelope for this geometry.
+    ///
+    /// See: [`OGR_G_GetEnvelope`](https://gdal.org/api/vector_c_api.html#_CPPv417OGR_G_GetEnvelope12OGRGeometryHP11OGREnvelope)
+    pub fn envelope(&self) -> Envelope {
+        let mut envelope = MaybeUninit::uninit();
+        unsafe {
+            gdal_sys::OGR_G_GetEnvelope(self.c_geometry(), envelope.as_mut_ptr());
+            envelope.assume_init()
+        }
+    }
+
+    /// Computes and returns the axis aligned 3D bounding envelope for this geometry.\
+    ///
+    /// See: [`OGR_G_GetEnvelope3D`](https://gdal.org/api/vector_c_api.html#_CPPv419OGR_G_GetEnvelope3D12OGRGeometryHP13OGREnvelope3D)
+    pub fn envelope3d(&self) -> Envelope3D {
```

Comment:
Nit:

```suggestion
    pub fn envelope_3d(&self) -> Envelope3D {
```

Not sure about this one, what do you think?